pkg/service/presentation/storage.go

```diff
@@ -140,7 +140,11 @@ func (ps *Storage) DeletePresentation(ctx context.Context, id string) error {
 }
 
 func (ps *Storage) StoreSubmission(ctx context.Context, s prestorage.StoredSubmission) error {
-	id := s.Submission.ID
+	sub, ok := s.VerifiablePresentation.PresentationSubmission.(exchange.PresentationSubmission)
+	if !ok {
```

Comment:
any pros and cons to doing it this way vs reflections vs json marshalling ?

Reply:
Synced offline. Casting guarantees that correct type is the one present.